Messaging.md

```diff
@@ -363,92 +371,210 @@ Witnesses should be sorted by the `input_serial_id` sent in `funding_input` defi
+   * [`u8`:`base`]
    * [`bool`:`is_signed`]
    * [`string`:`unit`]
    * [`int32`:`precision`]
    * [`u16`:`nb_digits`]
 
+### The `oracle_event_timestamp` Type
+
+This type contains timestamp information about when an `oracle_event` is to happen.
+
+There are two kinds of `oracle_event_timestamps`: fixed timestamps for events with a
+known (expected) epoch time at which the event will occur, and range for events whose
+time is not known ahead of time and also for events which may occur at any time in the
+specified range.
+
+When using an `oracle_event_timestamp_range` for events which may occur at any time,
+the `earliest_expected_time_epoch` should correspond to the time at which the oracle
+announcement was created and `latest_expected_time_epoch` should correspond to the
+latest time for which the oracle will be responsible for attesting to the event's occurence.
+Note that it is possible to have functionally open-ended events through the use of very large
+`latest_expected_time_epoch` timestamps.
+
+#### `fixed_oracle_event_timestamp`
+
+1. implements: `oracle_event_timestamp`
+1. type: 0
+1. data:
+   * [`u32`:`expected_time_epoch`]
+
+#### `range_oracle_event_timestamp`
+
+1. implements: `oracle_event_timestamp`
+1. type: 1
+1. data:
+   * [`u32`:`earliest_expected_time_epoch`]
+   * [`u32`:`latest_expected_time_epoch`]
+
 ### The `oracle_event` Type
 
 This type contains information provided by an oracle on an event that it will attest to.
 See [the Oracle specifications](./Oracle.md#oracle-event) for more details.
 
-**For backward compatibility reasons, this type is currently serialized as a TLV and uses u16 for collection prefix. It is expected to be changed to the new serialization format in a near future.**
-
 #### `oracle_event`
 
-1. type: 55330
 1. data:
-   * [`u16`:`nb_nonces`]
-   * [`nb_nonces*x_point`:`oracle_nonces`]
-   * [`u32`:`event_maturity_epoch`]
+   * [`oracle_event_timestamp`:`timestamp`]
    * [`event_descriptor`:`event_descriptor`]
    * [`string`:`event_id`]
 
+### The `oracle_metadata` Type
+
+This type contains static oracle information and can be used to import trusted oracles into wallets.
+This information should be saved by DLC nodes.
+
+#### `oracle_metadata`
+
+1. data:
+   * [`x_point`:`announcement_public_key`]
+   * [`string`:`oracle_name`]
+   * [`string`:`oracle_description`]
+   * [`u32`:`timestamp`]
+   * [`oracle_schemes`:`oracle_schemes`]
+   * [`signature`:`oracle_metadata_signature`]
+
+where the `oracle_metadata_signature` is a Schnorr signature by the `announcement_public_key` using the tag `oraclemetadata/v0` of the sha256 hash of the serialized `oracle_metadata` message omitting the `announcement_public_key` and of course the the `oracle_metadata_signature`.
+
+#### Rationale
+
+Separate public keys are required for announcement and attestation because the attestation scheme
+used in the current DLC spec reduces the security of non-attestation Schnorr signatures issued by the
```

Comment:
Any citation/reason why it degrades security? It seems we state this as truth without some sort of reason why.